### ISSUE TITLE
quote values of type String

### ISSUE THREAD

**Issue by close2** [close2/csv#6]
[ '1', 3 ] should result in '"1",3'
Currently 1 is not quoted.

**Comment by Zectbumo** [comment]
No, this is not correct. In CSV there is no difference between 1,3 and "1",3. There is no int and string type in CSV like you might be thinking.

**Comment by close2** [comment]
Agreed.

I've also received a bug report that numbers are not always quoted, which shouldn't be necessary.

I guess quoting behavior should be an enum: always,onlyStrings,onlyWhenNecessary (something like that)
This should be easy to implement, and make everyone happy.

Not sure about the default however.

**Comment by Zectbumo** [comment]
The enum would be nice. As for the default, I recommend quoting **only** when it is required. This will result in the smallest file size and will be easier to read the raw output.

**Comment by 926** [comment]
I also get :
"A value of type 'dynamic' can't be assigned to a variable of type 'String'."
In : lib/csv_settings_autodetection.dart
